packages/protocol/src/action-protocol/element-creation.ts

```diff
@@ -14,69 +14,159 @@
+            isOperation: true,
+            elementTypeId,
+            ...options
+        };
+    }
+}
 /**
- * In order to create an edge in the model the client can send a `CreateEdgeOperation` with the necessary information to create that edge.
+ * In order to create an edge in the model the client can send a `CreateEdgeOperation` with the necessary information to create
+ * the element that corresponds to that edge in the source model.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `CreateEdgeOperations`.
  */
-export class CreateEdgeOperation implements CreateOperation {
-    static readonly KIND = 'createEdge';
-
-    constructor(
-        public readonly elementTypeId: string,
-        public sourceElementId: string,
-        public targetElementId: string,
-        public args?: Args,
-        public readonly kind: string = CreateEdgeOperation.KIND
-    ) {}
-}
+export interface CreateEdgeOperation extends CreateOperation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof CreateEdgeOperation.KIND;
+
+    /*
+     * The source element.
+     */
+    sourceElementId: string;
 
-export function isCreateEdgeOperation(action: any): action is CreateEdgeOperation {
-    return (
-        isActionKind(action, CreateEdgeOperation.KIND) &&
-        isString(action, 'elementTypeId') &&
-        isString(action, 'sourceElementId') &&
-        isString(action, 'targetElementId')
-    );
+    /*
+     * The target element.
+     */
+    targetElementId: string;
 }
 
+export namespace CreateEdgeOperation {
+    export const KIND = 'createEdge';
+
+    export function is(object: any): object is CreateEdgeOperation {
+        return (
+            CreateOperation.hasKind(object, KIND) && hasStringProp(object, 'sourceElementId') && hasStringProp(object, 'targetElementId')
+        );
+    }
+
+    export function create(options: {
+        elementTypeId: string;
+        sourceElementId: string;
+        targetElementId: string;
+        args?: Args;
+    }): CreateEdgeOperation {
+        return {
+            kind: KIND,
+            isOperation: true,
+            ...options
+        };
+    }
+}
 /**
- * The client sends a `DeleteElementOperation` to the server to request the deletion of an element from the model.
+ * The client sends a `DeleteElementOperation` to the server to request the deletion of an element from the source model.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `DeleteElementOperations`.
  */
-export class DeleteElementOperation implements Operation {
-    static readonly KIND = 'deleteElement';
-    constructor(readonly elementIds: string[], public readonly kind: string = DeleteElementOperation.KIND) {}
+export interface DeleteElementOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.